### PR TITLE
support overrides in reference doc

### PR DESCRIPTION
**Pull request by superyyrrzz**
@vwxyzh @chenkennt @qinezh @ansyral @hellosnow

### CODE REVIEW

**Comment by hellosnow** [comment]
> 

minor: remove extra line below?

---
Refers to: src/Microsoft.DocAsCode.Build.ManagedReference/BuildOutputs/ApiTypeAndSpec.cs:15 in 47d61b6. [](commit_id = 47d61b62cb350b7d11c227162773b883a1b1a1b0, deletion_comment = False)

File: src/Microsoft.DocAsCode.Build.ManagedReference/BuildOutputs/ApiTypeAndSpec.cs [hunk 1]
```diff
@@ -20,5 +20,36 @@ public class ApiTypeAndSpec
...
+        {
+            return new ApiTypeAndSpec
+            {
+                Uid = uid,
+                Spec = ApiBuildOutputUtility.GetSpec(uid, references, supportedLanguages),
+                _needExpand = false,
+            };
+        }
+
+        public void Expand(Dictionary<string, ApiReferenceBuildOutput> references, string[] supportedLanguages)
```
Comment (hellosnow) [inline src/Microsoft.DocAsCode.Build.ManagedReference/BuildOutputs/ApiTypeAndSpec.cs]: consider to add IExpandable interface later?

File: src/Microsoft.DocAsCode.Build.ManagedReference/BuildOutputs/ApiTypeAndSpec.cs [hunk 2]
```diff
@@ -20,5 +20,36 @@ public class ApiTypeAndSpec
...
+        {
+            if (string.IsNullOrEmpty(uid)) return null;
+
+            return new ApiTypeAndSpec
+            {
+                Uid = uid,
+            };
+        }
+
+        public static ApiTypeAndSpec FromUid(string uid, Dictionary<string, ApiReferenceBuildOutput> references, string[] supportedLanguages)
```
Comment (hellosnow) [inline src/Microsoft.DocAsCode.Build.ManagedReference/BuildOutputs/ApiTypeAndSpec.cs]: also check for uid?

File: src/docfx.website.themes/default(zh-cn)/global.json [hunk 3]
```diff
@@ -17,6 +17,7 @@
   "namespace": "命名空间",
   "assembly": "程序集",
   "syntax": "语法",
+  "overrides": "替代",
```
Comment (vwxyzh) [inline src/docfx.website.themes/default(zh-cn)/global.json]: modify template in another pull request?

File: src/Microsoft.DocAsCode.Build.ManagedReference/BuildOutputs/ApiTypeAndSpec.cs (hunk 1 above)
Comment (superyyrrzz) [inline src/Microsoft.DocAsCode.Build.ManagedReference/BuildOutputs/ApiTypeAndSpec.cs]: Sounds good! add this later this sprint.

---
In reply to: [64332966](https://github.com/dotnet/docfx/pull/332#discussion_r64332966) [](ancestors = 64332966)

**Comment by hellosnow** [comment]
:shipit:

File: src/docfx.website.themes/default(zh-cn)/global.json (hunk 3 above)
Comment (qinezh) [inline src/docfx.website.themes/default(zh-cn)/global.json]: "重写"?  [https://msdn.microsoft.com/zh-cn/library/ebca9ah3.aspx](https://msdn.microsoft.com/zh-cn/library/ebca9ah3.aspx)

**Comment by superyyrrzz** [comment]
@qinezh @vwxyzh updated

**Comment by qinezh** [comment]
:shipit:

**Comment by vwxyzh** [comment]
👍

**Comment by superyyrrzz** [comment]
committed in https://github.com/dotnet/docfx/commit/f913c82ad97b5835ecce967cf082f7a8457cbf8d

**Pull request closed without merging**